plugins/teststeps/awsfileupload/awsFileUpload.go

```diff
@@ -0,0 +1,299 @@
+	// retrieve s3credprofile
+	ts.s3credprofile = params.GetOne("s3credprofile").String()
+
+	// Retrieving parameter as bool
+	// validate emit_urltoreport
+	param = params.GetOne("emit_urltoreport")
+	if !param.IsEmpty() {
+		v, err := strconv.ParseBool(param.String())
+		if err != nil {
+			return fmt.Errorf("invalid non-boolean `emit_urltoreport` parameter: %v", err)
+		}
+		ts.emitURLtoReport = v
+	}
+	// validate compress
+	param = params.GetOne("compress")
+	if !param.IsEmpty() {
+		v, err := strconv.ParseBool(param.String())
+		if err != nil {
+			return fmt.Errorf("invalid non-boolean `emit_urltoreport` parameter: %v", err)
+		}
+		ts.compress = v
+	}
+	return nil
+}
+
+// ValidateParameters validates the parameters associated to the TestStep
+func (ts *FileUpload) ValidateParameters(_ xcontext.Context, params test.TestStepParameters) error {
+	return nil
+}
+
+// New initializes and returns a new awsFileUpload test step.
+func New() test.TestStep {
+	return &FileUpload{}
+}
+
+// Load returns the name, factory and events which are needed to register the step.
+func Load() (string, test.TestStepFactory, []event.Name) {
+	return Name, New, Events
+}
+
+// createTarArchive creates compressed data writer and invokes addFileToArchive
+func createTarArchive(file string, buf bytes.Buffer) (bytes.Buffer, error) {
+	// Create gzip and tar writers
+	gzwriter := gzip.NewWriter(&buf)
+	defer gzwriter.Close()
+	tarwriter := tar.NewWriter(gzwriter)
+	defer tarwriter.Close()
+	// Write file into tar archive
+	err := addFileToArchive(tarwriter, file)
+	if err != nil {
+		return buf, err
+	}
+	return buf, nil
+}
+
+// addFileToArchive takes the data and writes it into the tar archive
+func addFileToArchive(tarwriter *tar.Writer, filename string) error {
+	// Open the file which shall be written into the tar archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	// Retrieve the file stats
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+	// Create a tar header from the file stats
+	header, err := tar.FileInfoHeader(info, info.Name())
+	if err != nil {
+		return err
+	}
+	header.Name = filename
+	// Write file header to the tar archive
+	err = tarwriter.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+	// Write the file into the tar archive
+	_, err = io.Copy(tarwriter, file)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Upload the file that is specified in the JobDescritor
+func (ts *FileUpload) upload(filename string, gzbuffer []byte) (string, error) {
+	// Create a single AWS session (we can re use this if we're uploading many files)
+	s, err := session.NewSession(&aws.Config{Region: aws.String(ts.s3region),
+		Credentials: credentials.NewSharedCredentials(
+			"", // your credential file path (default if empty)
+			"", // profile name (default if empty)
+		)})
+	if err != nil {
+		return "Could not open a new session.", err
```

Comment:
fix to 
```
return fmt.Errorf("could not open a new session: %w", err)
```

note that error strings in Go should never start with a capital letter